lib/orbit/evented.js

```diff
@@ -191,72 +191,36 @@ var Evented = {
       return listeners;
     },
 
-    resolve: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
-
-      return new Orbit.Promise(function(resolve, reject) {
-        var resolveEach = function() {
-          if (listeners.length === 0) {
-            reject();
-          } else {
-            var listener = listeners.shift();
-            var response = listener[0].apply(listener[1], args);
-
-            if (response) {
-              response.then(
-                function(success) {
-                  resolve(success);
-                },
-                function(error) {
-                  resolveEach();
-                }
-              );
-            } else {
-              resolveEach();
-            }
-          }
-        };
-
-        resolveEach();
-      });
+    resolve(eventNames, ...args) {
+      const listeners = this.listeners(eventNames);
+
+      function resolveEach([listener, ...listeners]) {
+        if(!listener) return;
+
+        return Orbit.resolve()
+          .then(() => {
+            const [callback, binding] = listener;
+            return callback.apply(binding, args);
+          })
+          .then(result => { return result || resolveEach(listeners); })
+          .catch(() => { return resolveEach(listeners); });
+      }
+
+      return resolveEach(listeners);
     },
 
-    settle: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
+    settle(eventNames, ...args) {
+      const listeners = this.listeners(eventNames);
 
-      return new Orbit.Promise(function(resolve) {
-        var settleEach = function() {
-          if (listeners.length === 0) {
-            resolve();
-          } else {
-            var listener = listeners.shift();
-            var response;
-            try {
-              response = listener[0].apply(listener[1], args);
-            } catch (e) {
-              console.error('Orbit ignored error in event listener', eventNames);
-              console.error(e.stack || e);
-            }
-
-            if (response) {
-              return response.then(
-                function(success) {
-                  settleEach();
-                },
-                function(error) {
-                  settleEach();
-                }
-              );
-            } else {
-              settleEach();
-            }
-          }
-        };
+      return listeners.reduce((chain, [callback, binding]) => {
+        return chain
+          .then(() => { return callback.apply(binding, args); })
+          .catch((e) => {
+            console.error('Orbit ignored error in event listener', eventNames);
```

Comment:
I feel like we should abstract this into `Orbit.error(e, 'Orbit ignored error in event listener', eventNames);`